src/_C015.ino

```diff
@@ -0,0 +1,127 @@
+#include <ArduinoJson.h>
+
+bool CPlugin_015(byte function, struct EventStruct *event, String &string)
+{
+  bool success = false;
+
+  switch (function)
+  {
+  case CPLUGIN_PROTOCOL_ADD:
+  {
+    Protocol[++protocolCount].Number = CPLUGIN_ID_015;
+    Protocol[protocolCount].usesMQTT = false;
+    Protocol[protocolCount].usesTemplate = false;
+    Protocol[protocolCount].usesAccount = false;
+    Protocol[protocolCount].usesPassword = false;
+    Protocol[protocolCount].usesID = false;
+    Protocol[protocolCount].defaultPort = 10051;
+    break;
+  }
+
+  case CPLUGIN_GET_DEVICENAME:
+  {
+    string = F(CPLUGIN_NAME_015);
+    break;
+  }
+
+  case CPLUGIN_PROTOCOL_SEND:
+  {
+    byte valueCount = getValueCountFromSensorType(event->sensorType);
+    C015_queue_element element(event);
+
+    MakeControllerSettings(ControllerSettings);
+    LoadControllerSettings(event->ControllerIndex, ControllerSettings);
+
+    for (byte x = 0; x < valueCount; x++)
+    {
+      element.txt[x] = formatUserVarNoCheck(event, x);
+    }
+    success = C015_DelayHandler.addToQueue(element);
+    scheduleNextDelayQueue(TIMER_C015_DELAY_QUEUE, C015_DelayHandler.getNextScheduleTime());
+    break;
+  }
+
+  case CPLUGIN_FLUSH:
+  {
+    process_c015_delay_queue();
+    delay(0);
+    break;
+  }
+  }
+  return success;
+}
+
+bool do_process_c015_delay_queue(int controller_number, const C015_queue_element &element, ControllerSettingsStruct &ControllerSettings)
+{
+
+  if (!WiFiConnected(10))
+  {
+    return false;
+  }
+
+  WiFiClient client;
+  if (!ControllerSettings.connectToHost(client))
+  {
+    connectionFailures++;
+    addLog(LOG_LEVEL_ERROR, String(F("ZBX: Cannot connect")));
+    return false;
+  }
+  statusLED(true);
+  if (connectionFailures)
+    connectionFailures--;
+
+  LoadTaskSettings(element.TaskIndex);
+
+  {
+
+    byte valueCount = getValueCountFromSensorType(element.sensorType);
+    // char itemNames[valueCount][2];
+    for (byte x = 0; x < valueCount; x++)
+    {
+
+
+const size_t capacity = JSON_ARRAY_SIZE(1) + JSON_OBJECT_SIZE(2) + JSON_OBJECT_SIZE(3);
+DynamicJsonBuffer jsonBuffer(capacity);
+
+JsonObject& root = jsonBuffer.createObject();
+root["request"] = "sender data";
+
+JsonArray& data = root.createNestedArray("data");
+
+JsonObject& data_0 = data.createNestedObject();
+data_0["host"] = Settings.Name;                            // zabbix hostname
+data_0["key"] = ExtraTaskSettings.TaskDeviceValueNames[x]; //valueName // zabbix item
+data_0["value"] = atof(element.txt[x].c_str());            
+    
+  char packet_header[] = "ZBXD\1";
+  char packet_content[capacity];
+  
+
+  root.printTo((char*)packet_content, root.measureLength()+1);
+  unsigned long long content_len = sizeof(packet_content);
+
+  //addLog(LOG_LEVEL_INFO, String(F("ZBX: ")) + packet_content);
+  ControllerSettings.connectToHost(client);
+  client.write(packet_header, sizeof(packet_header) - 1);
+  client.write((char *)&content_len, sizeof(content_len));
```

Comment:
I don't get this line.
`content_len` is an integer value (of 8 bytes size).
The result of `size_of` is very likely of type `size_t`, so why use an `uint64_t` type here?
And why print the content of that integer value in some binary format? (Endianness can also mess up what you try to send)

Reply:
The protocol wants it in 4 bytes (long long) (https://www.zabbix.com/documentation/4.2/manual/appendix/protocols/header_datalen). Int is 2 bytes.

Reply:
long long is 64 bit.
A long is 32 bit and I am quite sure int is also 32 bit.

When in doubt , use uint32_t.

Reply:
uint32_t is not working for me, uint64_t works, even thought the protocol asks for 4 bytes.